Domain/ProductManagement/Inventory.cs

```diff
@@ -0,0 +1,211 @@
+        private static InventoryUI _inventoryUI=new();
+        internal static void MainMenu()
+        {
+            _inventoryUI.DisplayMainMenu();
+            var userSelection=_inventoryUI.UserSelection();
+
+            switch (userSelection)
+            {
+                 case "1":
+                    AddANewProduct();
+                    break;
+                case "2":
+                    ViewAllProducts();
+                    break;
+                case "3":
+                    UpdateAProduct();
+                    break;
+                case "4":
+                    DeleteAProduct();
+                    break;
+                case "5":
+                    ViewAProduct();
+                    break;
+                default:
+                    _inventoryUI.PrintMessage("Invalid selection. Please try again.");
+                    break;
+            }
+        }
+        private static void AddANewProduct()
+        {
+            Product? product = null;
+
+            var noOfProducts = int.Parse(_inventoryUI.Input("How many Products do you want to add? "));
+
+            for (int i = 0; i < noOfProducts; i++)
+            {
+                var name = _inventoryUI.Input($"Enter the name of the {i + 1}. product you want to add: ");
+                
+                var price =double.Parse(_inventoryUI.Input("Enter the price of the product: "));
+
+                var quantity =int.Parse(_inventoryUI.Input("Enter the quantity you want to add: "));
+
+
+                //Checking if the product exists.
+                Product? existingProduct = _products.FirstOrDefault(p => p.ProductName == name);
+
+                if (existingProduct != null)
+                {
+                    //If the product  exists, you can't add one.
+                    _inventoryUI.PrintMessage("You can't add an existing product");
+                }
+                else
+                {
+                    //Creatng a new product if the product doesn't exists.
+                    if (quantity <= 150 && quantity > 0)
+                    {
+                        product = new Product(name, price, quantity);
+                        _products.Add(product);
+                        _inventoryUI.PrintMessage("Product added successfully.");
+                    }
+                    else if (quantity > 150)
+                        _inventoryUI.PrintMessage("You can't have more than 150.");
+                    else
+                        _inventoryUI.PrintMessage("You added zero items.");
+                }
+            }
+            MainMenu();
+        }
+        private static void ViewAllProducts()
+        {
+            if (_products.Count == 0)
+                _inventoryUI.PrintMessage("There is no Products in the Inventory yet.");
+            else
+            {
+                foreach (Product product in _products)
+                {
+                    _inventoryUI.PrintMessage(product.AllDetails());
```

Comment:
i like the idea of inventoryUI, but when we look at Inventory class, it stills coupled with the logic of UI. The methods in this class doesn't have to print to screen, for example this ViewAllProducts can return all products and the InventoryUI can print the list as wish. Inventory class can be used by anyone and can be used for different purposes. Lets think this way and update this class again ;)  Same thing if you have errors, you better maybe throw an exception here and the caller can do whatever with this exception (in your case, the caller will print a message to the screen, other callers might decide to log the error). So that's why this class still coupled by the UI.

Try also ChatGPT before, throw this code to it and ask for improvments ;)